### PR TITLE
feat(eap): Add new eap spans dataset

### PR DESCRIPTION
**Pull request by wmak**
- This adds a new dataset to the events endpoint called "spans" that will eventually replace the "spansIndexed" dataset
- Some of the column names are currently guesses as to what they will be in the new dataset, but we got the tests passing so lets try this out first and fix things as we go
- Note: timeseries queries are currently completely untested here
- Had to rewrite test_simple since the new table no longer stores statuses as codes

### CODE REVIEW

**Comment by volokluev** [comment]
What kind of feedback would you like on this PR from me?

**Comment by wmak** [comment]
@volokluev whatever you like, shared in the group chat just to update on progress of this dataset testing

File: src/sentry/utils/snuba.py [hunk 1]
```diff
@@ -172,6 +171,39 @@ def log_snuba_info(content):
...
+    "timestamp": "start_timestamp",
+    "trace": "trace_id",
+    "transaction": "segment_name",
+    "transaction.id": "segment_id",
+    "is_transaction": "is_segment",
+    "segment.id": "segment_id",
+    # We should be able to delete this and just use transaction
+    "origin.transaction": "segment_name",
+    # These are guesses:
+    "transaction.op": "attr_str[transaction_op]",
```
Comment (colin-sentry) [inline src/sentry/utils/snuba.py:202]: attr_str[transaction.op]

Comment (colin-sentry) [reply src/sentry/utils/snuba.py:202]: full list is this:

```
pub fn sentry_tag_key(&self) -> &str {
        match self {
            SpanTagKey::Release => "release",
            SpanTagKey::User => "user",
            SpanTagKey::UserID => "user.id",
            SpanTagKey::UserUsername => "user.username",
            SpanTagKey::UserEmail => "user.email",
            SpanTagKey::Environment => "environment",
            SpanTagKey::Transaction => "transaction",
            SpanTagKey::TransactionMethod => "transaction.method",
            SpanTagKey::TransactionOp => "transaction.op",
            SpanTagKey::Mobile => "mobile",
            SpanTagKey::DeviceClass => "device.class",
            SpanTagKey::BrowserName => "browser.name",
            SpanTagKey::SdkName => "sdk.name",
            SpanTagKey::SdkVersion => "sdk.version",
            SpanTagKey::Platform => "platform",

            SpanTagKey::Action => "action",
            SpanTagKey::AIPipelineGroup => "ai_pipeline_group",
            SpanTagKey::Category => "category",
            SpanTagKey::Description => "description",
            SpanTagKey::Domain => "domain",
            SpanTagKey::RawDomain => "raw_domain",
            SpanTagKey::Group => "group",
            SpanTagKey::HttpDecodedResponseContentLength => "http.decoded_response_content_length",
            SpanTagKey::HttpResponseContentLength => "http.response_content_length",
            SpanTagKey::HttpResponseTransferSize => "http.response_transfer_size",
            SpanTagKey::ResourceRenderBlockingStatus => "resource.render_blocking_status",
            SpanTagKey::SpanOp => "op",
            SpanTagKey::SpanStatus => "status",
            SpanTagKey::StatusCode => "status_code",
            SpanTagKey::System => "system",
            SpanTagKey::TimeToFullDisplay => "ttfd",
            SpanTagKey::TimeToInitialDisplay => "ttid",
            SpanTagKey::FileExtension => "file_extension",
            SpanTagKey::MainThread => "main_thread",
            SpanTagKey::CacheHit => "cache.hit",
            SpanTagKey::CacheKey => "cache.key",
            SpanTagKey::OsName => "os.name",
            SpanTagKey::AppStartType => "app_start_type",
            SpanTagKey::ReplayId => "replay_id",
            SpanTagKey::TraceStatus => "trace.status",
            SpanTagKey::MessagingDestinationName => "messaging.destination.name",
            SpanTagKey::MessagingMessageId => "messaging.message.id",
            SpanTagKey::ThreadName => "thread.name",
            SpanTagKey::ThreadId => "thread.id",
            SpanTagKey::ProfilerId => "profiler_id",
        }
    }
```

**Review by colin-sentry — CHANGES_REQUESTED** [review]

File: src/sentry/utils/snuba.py (hunk 1 above)
Comment (wmak) [reply src/sentry/utils/snuba.py:202]: @colin-sentry can you link where you pulled that from so I can reference it in the comments

Comment (wmak) [reply src/sentry/utils/snuba.py:202]: Wait why does this list have `transaction` i thought its called segment in this new table? Did that finally get fixed and its called transaction now?

Comment (colin-sentry) [reply src/sentry/utils/snuba.py:202]: https://github.com/getsentry/relay/blob/f2980a8bb45dbf69fc408d05615b022bb888518f/relay-event-normalization/src/normalize/span/tag_extraction.rs#L32

**Review by colin-sentry — APPROVED** [review]

**Pull request merged**